pkg/proxy/ipvs/proxier.go

```diff
@@ -1688,7 +1765,15 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 		curEndpoints.Insert(des.String())
 	}
 
-	for _, epInfo := range proxier.endpointsMap[svcPortName] {
+	endpoints := proxier.endpointsMap[svcPortName]
+
+	if !onlyNodeLocalEndpoints && utilfeature.DefaultFeatureGate.Enabled(kubefeatures.TopologyAwareServiceRouting) {
+		proxier.mu.Lock()
+		endpoints = proxy.FilterTopologyEndpoint(proxier.nodeName, proxier.nodeLabels, proxier.serviceMap[svcPortName].TopologyKeys(), endpoints)
```

Comment:
Structurally, I would prefer to see `EndpointFilter` interface that accepts some more generic inputs, then the proxier would have a list of filters to apply and call each one. We would install filters based on config/feature gates. Eventually we can implement additional filtering mechanisms then - for example something that looks at local latency metrics or something. That said, I don't think it's worth doing that at this point, it can wait until there is a need (if ever).

Reply:
Yes, we can consider extract a `EndpointFilter` interface to do generic filter logic when there is a need for that, but not worth doing that at this point